storage/innobase/fil/fil0crypt.cc

```diff
@@ -892,13 +892,13 @@ fil_crypt_start_encrypting_space(
 
 	if (space->encryption_type == Encryption::AES) {// We are re-encrypting space from MK encryption to RK encryption
 		crypt_data->encryption_rotation = Encryption::MASTER_KEY_TO_KEYRING;
-		ut_ad(space->encryption_key != NULL && space->encryption_iv != NULL);
```

Comment:
:+1:

Reply:
memory for space is allocated with ut_zalloc_nokey - can we add here :
static char zeros_test_block [ENCRYPTION_KEY_LEN];
memset (zeros_test_block, 0, sizeof testblock);
ut_ad(memcmp(space->encryption_key, zeros_test_block, ENCRYPTION_KEY_LEN) == 0); ?
space->encryption_iv in MySQL is ENCRYPTION_KEY_LEN, although the real size is 128 bits. We can skip checking it - encryption_key should be enough.

Reply:
I am testing
`#ifdef UNIV_DEBUG
		static const char all_zero_key[ENCRYPTION_KEY_LEN] = {0};
		ut_ad(memcmp(space->encryption_key, all_zero_key, ENCRYPTION_KEY_LEN) == 0);
#endif
`

Reply:
great, thanks